infra/modules/service/main.tf

```diff
@@ -41,11 +41,12 @@ locals {
 #-------------------
 
 resource "aws_ecs_service" "app" {
-  name            = var.service_name
-  cluster         = aws_ecs_cluster.cluster.arn
-  launch_type     = "FARGATE"
-  task_definition = aws_ecs_task_definition.app.arn
-  desired_count   = var.desired_instance_count
+  name                   = var.service_name
+  cluster                = aws_ecs_cluster.cluster.arn
+  launch_type            = "FARGATE"
+  task_definition        = aws_ecs_task_definition.app.arn
+  desired_count          = var.desired_instance_count
+  enable_execute_command = var.enable_service_execution ? true : null
```

Comment:
In our (OSRE) pattern, we provision a completely separate `cluster`, `service`, and `task_definition` for the jumphost. Then our `connect_to_jumphost.sh` starts up a task from the prebaked task def, and then does `start-session` to connect to it. 

Having people `start-session` directly into an app has some advantages and disadvantages. It will enable more fiddly and direct debugging within the app itself, at the tradeoff of potentially exposing access or secrets that weren't intended. It would also make it a little trickier to build a thorough audit trail of what's going on - I would expect a person running a session inside an app container _is_ the app from the perspective of IAM. 

Not disagreeing with doing it this way, just wanted to identify an alternative

Reply:
Totally fair and definitely true about the secrets (hence all the warnings I added for "don't do this in production!"). I've found this approach to be really valuable for debugging the actual deployment as-is.

That's a good question about the audit trail. Do you feel like that's necessary for lower environments?

Reply:
I'm far from an expert on the subject - but putting on my "good engineer" hat, I'd say that I really just want a strong audit trail anywhere there's real PII or real money involved. Which is _hopefully_ just prod... ;)

Reply:
... the terrifying idea of people putting real money or real PII in lower environments ... 😱